### PR TITLE
Remove temp incompatible checks

### PR DESCRIPTION
**Pull request by ron-murhammer**
Removes mostly null checks that were being used for save game compatibility

### CODE REVIEW

File: game-core/src/main/java/games/strategy/triplea/attachments/AbstractUserActionAttachment.java [hunk 1]
```diff
@@ -118,14 +118,6 @@ public void setCostResources(final IntegerMap<Resource> value) {
   }
 
   public IntegerMap<Resource> getCostResources() {
-    // TODO: remove for incompatible release
-    if (m_costResources == null) {
-      m_costResources = new IntegerMap<>();
-    }
-    if (m_costPU != 0) {
```
[the remark this answers is not in the document]
Comment (ron-murhammer) [reply game-core/src/main/java/games/strategy/triplea/attachments/AbstractUserActionAttachment.java]: Yeah, keeping that separate for now. Just hitting the incompatible checks. I'm actually reviewing this old issue and map PRs right now: https://github.com/triplea-game/triplea/issues/3475

File: game-core/src/main/java/games/strategy/triplea/delegate/BattleTracker.java [hunk 2]
```diff
@@ -87,8 +84,9 @@
   private BattleRecords m_battleRecords = null;
   // to keep track of all relationships that have changed this turn
   // (so we can validate things like transports loading in newly created hostile zones)
-  private final Collection<Tuple<Tuple<PlayerID, PlayerID>, Tuple<RelationshipType, RelationshipType>>> m_relationshipChangesThisTurn =
-      new ArrayList<>();
+  private final Collection<
+      Tuple<Tuple<PlayerID, PlayerID>, Tuple<RelationshipType, RelationshipType>>> m_relationshipChangesThisTurn =
+          new ArrayList<>();
```
[the remark this answers is not in the document]
Comment (ron-murhammer) [reply game-core/src/main/java/games/strategy/triplea/delegate/BattleTracker.java]: That's actually what the auto-formatter in eclipse did.

File: game-core/src/main/java/games/strategy/triplea/delegate/BattleTracker.java [hunk 3]
```diff
@@ -165,7 +163,8 @@ boolean didAllThesePlayersJustGoToWarThisTurn(final PlayerID p1, final Collectio
   private boolean didThesePlayersJustGoToWarThisTurn(final PlayerID p1, final PlayerID p2) {
     // check all relationship changes that are p1 and p2, to make sure that oldRelation is not war,
     // and newRelation is war
-    for (final Tuple<Tuple<PlayerID, PlayerID>, Tuple<RelationshipType, RelationshipType>> t : m_relationshipChangesThisTurn) {
+    for (final Tuple<Tuple<PlayerID, PlayerID>,
+        Tuple<RelationshipType, RelationshipType>> t : m_relationshipChangesThisTurn) {
```
[the remark this answers is not in the document]
Comment (ron-murhammer) [reply game-core/src/main/java/games/strategy/triplea/delegate/BattleTracker.java]: Same.

File: game-core/src/main/java/games/strategy/triplea/delegate/BattleTracker.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (ssoloff) [reply game-core/src/main/java/games/strategy/triplea/delegate/BattleTracker.java]: Interesting.  Did you upgrade Eclipse or create a new workspace recently?  Just wondering if you're still using the current TripleA formatter definition.

[the remark this answers is not in the document]
Comment (ron-murhammer) [reply game-core/src/main/java/games/strategy/triplea/delegate/BattleTracker.java]: I don't think so. I'm still on Eclipse 4.7 and haven't touched the formatter settings in a long time.

[the remark this answers is not in the document]
Comment (ssoloff) [reply game-core/src/main/java/games/strategy/triplea/delegate/BattleTracker.java]: Well, not going to hold up this PR for that.  When you have time, would you mind exporting your formatter settings and posting them to a gist or something?  I'd like to see what the difference is that's causing your instance to wrap this line and my instance to leave it untouched.

**Comment by ssoloff** [comment]
Thanks, @ron-murhammer.

Looks like, somehow, your line-wrapping settings for generic type parameters are different than what's in the repo (I trimmed irrelevant changes from the diff below):

```diff
$ diff <(xmllint --c14n .eclipse/format/triplea_java_eclipse_format_style.xml) <(xmllint --c14n ~/Downloads/TripleA_Formatter.xml)
61c60
< <setting id="org.eclipse.jdt.core.formatter.alignment_for_parameterized_type_references" value="0"></setting>
---
> <setting id="org.eclipse.jdt.core.formatter.alignment_for_parameterized_type_references" value="16"></setting>
113c111
< <setting id="org.eclipse.jdt.core.formatter.alignment_for_type_parameters" value="0"></setting>
---
> <setting id="org.eclipse.jdt.core.formatter.alignment_for_type_parameters" value="16"></setting>
218c215
< <setting id="org.eclipse.jdt.core.formatter.alignment_for_type_arguments" value="0"></setting>
---
> <setting id="org.eclipse.jdt.core.formatter.alignment_for_type_arguments" value="16"></setting>
```

I'm okay with the settings you're using, as they will solve the remaining Checkstyle line length violations.

Since you're still on Oxygen, would you like to submit a PR to update _.eclipse/format/triplea_java_eclipse_format_style.xml_?  Apparently, the schema for this document has changed (from version 13 to 14) in Photon.  So, if I do it, you might not be able to import it.  We'll probably need to sanity check the change to ensure it doesn't try to reformat the whole codebase. :smile:

(BTW, looks like our formatter XML hasn't been updated in quite some time, so you will see a ton of changes due to defaults being written for new settings introduced in the past couple of releases.)

**Pull request merged**